data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -43,4 +54,13 @@ public void output(Collection<T> records) {
      * @param records Records to be output
      */
     public abstract void doOutput(Collection<T> records);
+
+    @Override
+    public void shutdown() {
+        retryThread.stop();
+    }
+
+    public Thread.State getRetryThreadState() {
```

Comment:
Do we need this? If it is only for testing, please make it package protected (remove the `public` modifier).

Reply:
Yes, it is only for testing. Will remove the "public"